agent/lib/broker_controller.js

```diff
@@ -383,40 +427,58 @@ function translate(addresses_in, excluded_names, excluded_types) {
     return addresses_out;
 }
 
+BrokerController.prototype.update_address_setting = function (a) {
+    var self = this;
+    return self.broker.getAddressSettings(a.address).then(function (orig_settings) {
+
+        return self.get_address_settings(a).then(function (new_settings) {
+            if (compare_address_settings(orig_settings, new_settings) != 0) {
+                self.broker.removeAddressSettings(a.address).then(function () {
+                    return self.broker.addAddressSettings(a.address, new_settings);
+                });
+            }
+        });
+    });
+}
+
 BrokerController.prototype.delete_address = function (a) {
     var self = this;
-    if (a.type === 'queue') {
-        log.info('[%s] Deleting queue "%s"...', self.id, a.address);
-        return self.broker.destroyQueue(a.address).then(function () {
-            log.info('[%s] Deleted queue "%s"', self.id, a.address);
-            self.post_event(myevents.address_delete(a));
-        }).catch(function (error) {
-            log.error('[%s] Failed to delete queue %s: %s', self.id, a.address, error);
-            return self.broker.deleteAddress(a.address).then(function () {
-                log.info('[%s] Deleted anycast address %s', self.id, a.address);
+
+    return self.broker.getAddressSettings(a.address).then(function (settings) {
```

Comment:
why do we need to lookup the address settings here?  also, if the address settings are missing is it correct to not delete the queues etc?